### PR TITLE
Changes from pdk convert

### PR DESCRIPTION
**Pull request by danielparks**
This only contains changes from `pdk convert`, metadata updates, and changes to get the module to pass linting. Other PRs will clean other things up.

Note that there are two commits — one is just `pdk convert`, and the other has the lint fixes.

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -8,11 +8,15 @@
 ) inherits ssh::params {
   include ssh
 
-  if $::kernel == 'Linux' and ! defined(Package[$server_package]) {
-    package { $server_package:
-      ensure => latest,
-      notify => Service['sshd'],
+  if $facts['kernel'] == 'Linux' {
+    if ! defined(Package[$ssh::params::server_package]) {
```
Comment (genebean) [inline manifests/server.pp]: @danielparks if we are going to keep doing this lets change this to use `ensure_packages` from stdlib.

**Review by genebean — APPROVED** [review]

**Comment by genebean** [comment]
@danielparks I enabled travis-ci for this repo... please repush so that it will trigger the first build. FWIW, I have been known to do a `git commit --amend -a --no-edit ; git push -f` to accomplish that.

File: manifests/server.pp (hunk 1 above)
Comment (danielparks) [reply manifests/server.pp]: Oh, thanks for pointing that out. I thought `ensure_packages` broke when the parameters are different, but experimentation confirms that it doesn't.

**Comment by danielparks** [comment]
I added trivial unit tests (just try to compile `ssh` and `ssh::server` without params), and found two missing variables in `ssh::params`.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'spec_helper'
+
+describe 'ssh' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:params) { {} }
```
Comment (genebean) [inline spec/classes/init_spec.rb:6]: AFAIK this is unneeded.

File: spec/classes/server_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'spec_helper'
+
+describe 'ssh::server' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:params) { {} }
```
Comment (genebean) [inline spec/classes/server_spec.rb:6]: AFAIK this is unneeded.

File: spec/classes/init_spec.rb (hunk 2 above)
Comment (danielparks) [reply spec/classes/init_spec.rb:6]: Just confirmed, and you are correct. Removed.

**Pull request merged**